backend/workflow_manager/file_execution/models.py

```
@@ -120,32 +120,30 @@ def __str__(self):
 
     def update_status(
         self,
-        status: ExecutionStatus,
+        status: ExecutionStatus | str,
         execution_error: str = None,
```

Ruff (0.13.1)

PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)